### DISCUSSION TITLE
bullmq with expressjs how to handle queues and workers

### DISCUSSION THREAD

**Discussion by lord007tn** [taskforcesh/bullmq#803] [Q&A]
Hello guys 

so I have a rest API with expressjs 
I made a file called queues js contained all queues in the project 
in some API routes, I add jobs to the queue

the problem here is I'm running another script besides "index.js" called "workers.js" to run all workers and make them listen to the job added to the queues 

I don't know if this is the right way to do it, especially that I'm using docker in production so i don't know how to handle multiple server entry points in production.

any suggestion will help

**Comment by manast** [comment]
There is no right or wrong way and depends mostly on your requirements. My advice is, start with something that works and then you will figure out architectures that may work better for your use cases.

**Reply by lord007tn** [reply]
if you use another approach guide me, as I searched on web and that's the best I can find
